### DISCUSSION TITLE
Add buttons to command bar

### DISCUSSION THREAD

**Discussion by Spiffo** [rubberduck-vba/Rubberduck#6018] [Ideas]
Would it be possible to add the functions that are currently available under the right click on code to the Rubberduck Command bar? That way we would have much quicker access to them. Among other things, I use the Indent function a lot, and at the moment this has me going through right-click > click > click > click, with pointing to the correct option. It would be great if this could be done directly from the toolbar that has the Refresh button the path and the number of references, or in another toolbar.
Thanks for considering

**Comment by Greedquest** [comment]
@Spiffo `Ctrl`+ `'` - that's an apostrophe - to refresh and `Ctrl` + `Shift` + `m` to indent the current module are good to know. Not that I'm against making the buttons more accessible, but I personally rarely use the buttons now.

PS I often find the RD keyboard shortcuts become disconnected if the editor loses focus. That's okay though, as a workaround I do `Ctrl` + `E` then `Esc` to quickly bring up the export dialogue and close it immediately after, this gets the RD shortcuts working again.

PPS all RD keyboard shortcuts are configurable in the settings menu

**Reply by Greedquest** [reply]
Discoverability: add the text `Default: Ctrl+M` or whatever as a tooltip.

 - it's a quick win to introduce the concept of keyboard shortcuts
 - it's easy to implement default text, if a user changes the shortcut to something else you ideally want to show that on the tooltip but that feels more complicated and stateful. The user probably also knows where to find the menu if they have previously used it.
 - the phrasing "default" implies the existence of configurable shortcuts so will improve discoverability of that feature too - simply displaying the current registered shortcut may be worse in that regard, as well as harder to implement
 
I've personally never felt the need to stray from the default shortcuts, I think they were well chosen good job! But important to keep for accessibility (reduced dexterity) and localisation (different keyboard layouts).

**Comment by Spiffo** [comment]
Thanks for the pointers towards the keyboard shortcuts. I'm more of a "mouse-man" myself, so if additional buttons could still be considered...

**Reply by retailcoder** [reply]
The problem is that the toolbar also doubles as a status bar, and when you use `'@Description` annotations a lot the "info" label width is (annoyingly) very variable, so additional buttons would either move valuable context-sensitive information out of sight, or buttons would be constantly moving around... which is even worse I think.

That said the "find all references" button that _is_ there is indeed my personal go-to for that command, so I'd be more inclined to perhaps add a completely different, optional toolbar to do this... The only issue with that is that the command bar is contributing to tear-down instabilities, so adding another one isn't risk-free in terms of regression and overall stability.

Moreover, work has started on the v3 editor (Rubberduck is getting its own IDE-inside-the-VBE in v3, see project blog for details), and the old Office command bar is disappearing altogether in favor of an actual status bar... The good news is that a fully managed (.net) toolbar is totally happening in v3, so I think the best course of action would be to defer this issue to v3.

Side note, all hotkeys can be configured in the settings dialog 😉

**Reply by Greedquest** [reply]
> Side note, all hotkeys can be configured in the settings dialog 😉

@retailcoder Tangential - what do you think of adding a tooltip to the menu items with _default_ shortcut - e.g. `default: ctrl+m`. Seems like a quick win to improve discoverability without the complexity of syncing tooltips to the current hotkey settings.

**Reply by retailcoder** [reply]
In principle the hotkey combo is supposed to be showing in a dedicated area of the menu items, but it was lost somewhere along the way.

If tooltips can be added to Office command bar menu items (?), then I'm all for it - but this is going to be much simpler with the xaml menus in v3.

**Reply by Greedquest** [reply]
Tooltip is the wrong word. More like this I mean:

<img width="203" alt="image" src="https://user-images.githubusercontent.com/25348920/212338959-fe4ad7e3-d43a-4aa6-86d2-468bb06b6843.png"> 
<img width="256" alt="image" src="https://user-images.githubusercontent.com/25348920/212339180-2c2ff3ba-0925-42b6-892a-5d88952621b0.png">
